### PR TITLE
fix(RHINENG-1457): hsp dropdown not displaying hsps

### PR DESCRIPTION
**Pull request by johnsonm325**
Description:
I have a system with 4 hsps.  In the comparison modal, when I click on the hsp icon, it reports "There are no historical profiles to display".  See screenshot which shows in the Dev Tools window that the frontend is successfully retrieving the 4 hsps from the backend.

Steps to Reproduce:
1. Navigate to Operations -> Drift -> Comparison
2. Click Add systems or baselines
3. For a system that has hsps, click on the hsp icon. It reports no hsps. *Make sure via the Network tab in dev tools that the hsps fetch actually fetched hsps*

How reproducible: (Always / Intermittent / Random) Always

Actual Results:
No hsps reported

Expected results:
A popup window that allows me to select hsps

## Secure Coding Practices Checklist Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

**Comment by johnsonm325** [comment]
In light of the urgency of this issue and the fact that I would need to rework a lot of tests, I commented out and skipped a lot of tests. I can come back and redo these tests at a later date.

File: src/SmartComponents/HistoricalProfilesPopover/HistoricalProfilesCheckbox/HistoricalProfilesCheckbox.js [hunk 1]
```diff
@@ -25,43 +20,36 @@ export class HistoricalProfilesCheckbox extends Component {
...
-    handleChange = () => {
-        const { checked } = this.state;
-        const { onSelect, profile } = this.props;
+    };
 
-        this.setState({
-            checked: !checked
-        });
+    const [ checked, setChecked ] = useState(findChecked() || false);
+    const isFirstRender = useRef(true);
```
Comment (bastilian) [inline src/SmartComponents/HistoricalProfilesPopover/HistoricalProfilesCheckbox/HistoricalProfilesCheckbox.js:26]: @johnsonm325 Can you elaborate on this. That doesn't feel right.

**Review by AsToNlele — CHANGES_REQUESTED** [review]
This works when there are Historical profiles, but when there are none, I see 404 in the network tab, but in the UI this keeps loading
image

**Comment by petrblaho** [comment]
I added some notes to the original ticket about the experienced behaviour (could help with testing and finding the issue).

**Comment by johnsonm325** [comment]
@AsToNlele you can take another look at this. Your problem should be fixed.

**Review by AsToNlele — APPROVED** [review]
Sorry about the delay,
LGTM :smile:

**Pull request merged**